packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,223 @@
+    Name: 'canariescanary8ee009f',
+  });
+});
+
+test('Name validation does not fail when using Tokens', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: Lazy.stringValue({produce: () => 'myCanary'}),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN: no exception
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary');
+});
+
+test('Throws when name is specified incorrectly', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'myCanary',
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  }))
+    .toThrowError('Canary name must be lowercase, numbers, hyphens, or underscores (no spaces) (myCanary)');
+});
+
+test('Throws when name has more than 21 characters', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'a'.repeat(22),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  }))
+    .toThrowError('Canary name is too large, must be <= 21 characters, but is 22');
+});
+
+test('An existing role can be specified instead of auto-created', () => {
+  // GIVEN
+  const role = new iam.Role(stack, 'role', {
+    assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+  });
+
+  // role.addToPolicy(/* required permissions per the documentation */);
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    role,
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    ExecutionRoleArn: stack.resolve(role.roleArn),
+  });
+});
+
+test('An existing bucket can be specified instead of auto-created', () => {
+  // GIVEN
+  const bucket = new s3.Bucket(stack, 'mytestbucket');
+  const prefix = 'canary';
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    artifactsBucketLocation: { bucket, prefix },
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    ArtifactS3Location: stack.resolve(bucket.s3UrlForObject(prefix)),
+  });
+});
+
+test('Runtime can be specified', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    runtime: synthetics.Runtime.SYNTHETICS_1_0,
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Runtime can be customized', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    runtime: new synthetics.Runtime('syn-1.0'),
```

Comment:
You're not checking anything since your "custom" runtime value is the same as the normal one. Put some fun string here instead... For example `fancy-future-runtime-1337.42`